control-plane/catalog/to-consul/resource.go

```diff
@@ -641,6 +667,20 @@ func (t *ServiceResource) generateRegistrations(key string) {
 	}
 }
 
+// Sets the passing service weight
+func (t *ServiceResource) setServiceWeight(weight string, appsCount int, r consulapi.CatalogRegistration) error {
+	// error validation if the input param is a number
+	weightI, err := strconv.Atoi(weight)
+	if err != nil {
+		return err
+	} else if weightI > 1 {
+		r.Service.Weights = consulapi.AgentWeights{
+			Passing: weightI / appsCount,
```

Comment:
Lets say weight is 10 and we have 3 service - 
Should we set weights 4, 3, 3 or 3, 3, 3?
Later one sums to 9 not 10.

Reply:
We can certainly try for 4,3,3 but 3,3,3 is also near accurate. In my experience, the traffic routing is not so precise.